lib/active_record/connection_adapters/spanner/schema_statements.rb

```diff
@@ -74,12 +74,17 @@ def drop_table table_name, options = {}
           execute_schema_statements statements
         end
 
+        # Creates a join table that uses all the columns in the table as the primary key.
```

Comment:
I've added some additional documentation to the code. The reason for this is:
1. ActiveRecord will by default generate join tables __without__ a primary key
2. Cloud Spanner requires __all__ tables to have a primary key

Instead of adding an additional column to all join tables that only has the purpose of being the primary key, the adapter will use all the columns in a join table for the primary key definition. This is possible as all values in a join table should be unique anyways (there's no point in adding the same relation between two records twice).

Reply:
Got it. Thanks for the doc.